pkg/imgutil/imgutil.go

```diff
@@ -373,3 +387,61 @@ func ParseRepoTag(imgName string) (string, string) {
 
 	return repository, tag
 }
+
+func ParseFilters(filters []string) ([]string, []string, error) {
+	var beforeFilters []string
+	var afterFilters []string
+	for _, filter := range filters {
+		tempFilterToken := strings.Split(filter, "=")
+		switch len(tempFilterToken) {
+		case 1:
+			return nil, nil, fmt.Errorf("invalid filter %q", filter)
+		case 2:
+			if tempFilterToken[0] == "before" {
+				canonicalRef, err := referenceutil.ParseAny(tempFilterToken[1])
+				if err != nil {
+					return nil, nil, err
+				}
+				beforeFilters = append(beforeFilters, fmt.Sprintf("name==%s", canonicalRef.String()))
+				beforeFilters = append(beforeFilters, fmt.Sprintf("name==%s", tempFilterToken[1]))
+			} else if tempFilterToken[0] == "since" {
+				canonicalRef, err := referenceutil.ParseAny(tempFilterToken[1])
+				if err != nil {
+					return nil, nil, err
+				}
+				afterFilters = append(afterFilters, fmt.Sprintf("name==%s", canonicalRef.String()))
+				afterFilters = append(afterFilters, fmt.Sprintf("name==%s", tempFilterToken[1]))
+			} else {
+				return nil, nil, fmt.Errorf("invalid filter %q", filter)
+			}
+		default:
+			return nil, nil, fmt.Errorf("invalid filter %q", filter)
+		}
+	}
+	return beforeFilters, afterFilters, nil
+}
+
+func FilterImages(labelImages []images.Image, beforeImages []images.Image, afterImages []images.Image) []images.Image {
+	sort.Slice(beforeImages, func(i, j int) bool {
```

Comment:
I think we do not need sort as we need to traverse the entire list anyway.

Reply:
maybe max or min is enough?